### ISSUE TITLE
Clearing the ResourceSet in Xtext Builder has bad performance.

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#525]
org.eclipse.xtext.builder.clustering.ClusteringBuilderState.clearResourceSet(ResourceSet)

does disable notification for the resourceset, but not for the resources. so notifications for the inverse will be calculated anyway.

same for

org.eclipse.xtext.builder.impl.XtextBuilder.doBuild(ToBeBuilt, IProgressMonitor, BuildType)

i also gonna consult ed on this
https://www.eclipse.org/forums/index.php/t/1091284/

**Comment by cdietrich** [comment]
``` 
package org.xtext.example.mydsl.ide;

import org.eclipse.emf.common.util.URI;
import org.eclipse.emf.ecore.resource.Resource;
import org.eclipse.xtext.resource.SynchronizedXtextResourceSet;
import org.xtext.example.mydsl.MyDslStandaloneSetup;

import com.google.inject.Injector;

public class Main {
	
	public static void main(String[] args) {
		Injector injector = new MyDslStandaloneSetup().createInjectorAndDoEMFRegistration();
		SynchronizedXtextResourceSet rs = injector.getInstance(SynchronizedXtextResourceSet.class);
		rs.eSetDeliver(false);
		for (int i = 0; i < 100000;i++) {
			if (i%1000==0) {
				System.out.println(i);
			}
			rs.createResource(URI.createURI("dummy_"+i+".mydsl"));
		}
		System.out.println("clearers gonna clear");
		rs.eSetDeliver(false);
		for (Resource r : rs.getResources()) {
			r.eSetDeliver(false);
		}
		rs.getResources().clear();
		rs.eAdapters().clear();
		System.out.println("done");
	}

}
```

**Comment by cdietrich** [comment]
i am now sure how the potential fix interfers with onchangeEvicting caches

**Comment by cdietrich** [comment]
Fixed in 2.14
